src/components/ListItem.jsx

```diff
@@ -1,5 +1,64 @@
 import './ListItem.css';
+import { useState, useEffect } from 'react';
+import { updateItem } from '../api/firebase';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+const milliSecondsInADay = 24 * 60 * 60 * 1000;
+const currentTimeInMilliseconds = Date.now();
+
+export function ListItem({ listToken, item }) {
+	let { id, name, isChecked, dateLastPurchased, totalPurchases } = item;
+
+	const [isPurchased, setIsPurchased] = useState(isChecked);
+
+	const dateLastPurchasedInMilliseconds = dateLastPurchased
+		? dateLastPurchased.seconds * 1000
+		: null;
+
+	const timeElasped =
+		currentTimeInMilliseconds - dateLastPurchasedInMilliseconds;
+
+	useEffect(() => {
+		if (timeElasped >= milliSecondsInADay) {
+			const items = {
+				isChecked: false,
+			};
+			updateItem(listToken, id, items);
+			setIsPurchased(false);
+		}
+	}, [listToken, timeElasped, id]);
+
+	const handleCheckboxChange = (e) => {
+		if (isPurchased) {
+			const items = {
+				isChecked: false,
+			};
+			setIsPurchased(false);
+			updateItem(listToken, id, items);
+		} else {
+			const count = totalPurchases + 1;
+			const items = {
+				isChecked: true,
+				dateLastPurchased: new Date(),
+				totalPurchases: count,
+			};
+
+			updateItem(listToken, id, items);
+			setIsPurchased(true);
+		}
+	};
+
+	return (
+		<li className="ListItem" key={id}>
+			<label htmlFor="listeItem">
+				<input
+					type="checkbox"
+					id="id"
+					name="id"
+					checked={isPurchased}
+					onChange={handleCheckboxChange}
+				/>
+				{name}
+			</label>
```

Comment:
```suggestion
			<label>
				<input
					type="checkbox"
					name="id"
					checked={isPurchased}
					onChange={handleCheckboxChange}
				/>
				{name}
			</label>
```

It looks like there's a small typo here ("listeItem"), but I think we can actually remove the `htmlFor` attribute, since we have the input and item name nested inside the label. From [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label):

> Alternatively, you can nest the <input> directly inside the <label>, in which case the for and id attributes are not needed because the association is implicit